iFixit/checkout-ui-extensions
# Determine iFixit base url using current store

Uses ifixit.com in production always, uses cominor.com in dev unless a developer subdomain is present in the settings for an app block.

## Motivation

Shopify offers nothing analogous to environment variables or global theme settings for checkout extensions. See [these 3 issues](https://github.com/Shopify/ui-extensions/issues?q=is%3Aissue+environment+variable+is%3Aopen). The settings defined in the toml file will vary at each installed app block in the checkout editor. That means the newsletter component could have different settings values than the VAT component, for example.

After searching around online, the strategy other developers are using is either hardcoding values based on the current shop (similar to this implementation), or using shop metafields to expose values.

It seemed laborious to add a shop metafield on each store, so I chose a hardcoded solution.

## QA

- Run local dev with `pnpm dev`
- Make sure that the newsletter signup appears and works correctly on the order status page.
